ProfessorAssessor/Networking/Networker.m

```diff
@@ -37,34 +39,100 @@ + (void)createCourseWithName:(NSString *)name
+                                     completion:(void(^)(BOOL succeeded,
+                                                         NSError *_Nullable error))completion {
+    Review *newReview = [Networker createReviewWithCourse:course
+                                                professor:professor
+                                                  content:content
+                                                   rating:rating];
+
+    PFQuery *query = [Review query];
+
+    [query whereKey:@"professor" equalTo:professor];
+
+    [newReview saveInBackgroundWithBlock:^(BOOL succeeded,
+                                           NSError *_Nullable error) {
+        if (error == nil) {
+            [query
+             countObjectsInBackgroundWithBlock:^(int number,
+                                                NSError *_Nullable error) {
+                if (error == nil) {
+                    professor.averageRating = [Networker
+                                               updateRatingWithDouble:[rating doubleValue]
+                                               forProfessor:professor
+                                               reviewCount:number];
+
+                    [professor
+                     saveInBackgroundWithBlock:^(BOOL succeeded,
+                                                 NSError *_Nullable error) {
+                        if (error == nil) {
+                            [Networker addReview:newReview
+                                        toCourse:course
+                                  withCompletion:completion];
+                        }
+                    }];
+                }
+            }];
+        }
+    }];
+}
+
++ (Review *)createReviewWithCourse:(Course *)course
+                         professor:(Professor *)professor
+                           content:(NSString *)content
+                            rating:(NSNumber *)rating {
     Review *newReview = [Review new];
-    newReview.reviewer = (User *)[PFUser currentUser];
+
+    newReview.reviewer = [User currentUser];
     newReview.course = course;
     newReview.rating = rating;
     newReview.content = content;
     newReview.professor = professor;
 
-    [newReview saveInBackgroundWithBlock:^(BOOL succeeded, NSError *_Nullable error) {
-        if (error == nil) {
-            PFRelation *reviews = [course relationForKey:@"reviews"];
-            [reviews addObject:newReview];
-            [course saveInBackgroundWithBlock:completion];
-        }
-    }];
+    return newReview;
+}
+
++ (NSNumber *)updateRatingWithDouble:(double)newRating
+                        forProfessor:(Professor *)professor
+                         reviewCount:(int)number{
+    double oldAverageRating = [professor.averageRating doubleValue];
+    double newAverageRating = oldAverageRating + ((newRating - oldAverageRating) / (double)number);
+    NSNumber *averageRatingAsNumber = [NSNumber numberWithDouble:newAverageRating];
```

Comment:
nit: This doesn't need to call out the type in the name. Can just be `averageRating`